File: SWE/LeetCode/0242_ValidAnagram/0242_ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Time: O(n + m), Space: O(n + m)
        s_d = {}
        for i in s:
            if i not in s_d:
                s_d[i] = 0
            s_d[i] += 1
        t_d = {}
        for i in t:
            if i not in t_d:
                t_d[i] = 0
            t_d[i] += 1
        return s_d == t_d

File: SWE/LeetCode/0242_ValidAnagram/0242_ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        // Time: O(n + m), Space: O(n + m)
        unordered_map<char,int> s_map, t_map;
        for (int i = 0; i < s.size(); i++) {
            s_map[s[i]]++;
        }
        for (int i = 0; i < t.size(); i++) {
            t_map[t[i]]++;
        }
        return s_map == t_map;
    }
};